src/meta_memcache/base/base_cache_pool.py

```diff
@@ -37,11 +37,10 @@ def __init__(
         self,
         serializer: BaseSerializer,
         binary_key_encoding_fn: Callable[[Key], bytes],
-        write_failure_tracker: Optional[BaseWriteFailureTracker] = None,
     ) -> None:
         self._serializer = serializer
         self._binary_key_encoding_fn = binary_key_encoding_fn
-        self._write_failure_tracker: Final = write_failure_tracker
+        self.OnWriteFailure: Final = WriteFailureEvent()
```

Comment:
Should this better be `on_write_failure`?

I typically prefer asking for a preconfigured `WriteFailureEvent` object in the constructor, as a param, and make it a private property. I think it is more discoverable, and it is not something you typically have to change after having the client instantiated. But no strong feelings, we can also keep it as is.

Reply:
👍 on the name.

Can you give a little more of an example for the other part?

Reply:
The other part is not really that important :) but to elaborate:

Discoverable: It is easier to spot this feature if in the constructor you have an optional `write_failure_event_tracker` argument, than by looking at the properties of the cache pool and discovering that you can add handlers that way.
Less error prone: I have found myself struggling with weird races: handlers were registered after the problem and not triggering, handlers were not being registered because that part of the code wasn't running XDD. When you do things on constructor is harder to shoot you on the foot, there is no extra runtime state to worry about.

Again for something as small and simple as this, don't bother. It was just a comment.